src/java/org/apache/cassandra/config/Config.java

```diff
@@ -178,6 +178,8 @@
     // (it takes much longer than 30s) as of Linux 4.12. If you want something that high set this to 0
     // (which picks up the OS default) and configure the net.ipv4.tcp_retries2 sysctl to be ~8.
     public int internode_tcp_user_timeout_in_ms = 30000;
+    // Similar to internode_tcp_user_timeout_in_ms but used specifically for streaming connection.
+    public int internode_streaming_tcp_user_timeout_in_ms = 0;
```

Comment:
> What does it mean for this timeout to default to zero instead of 30000 like its counterpart above?

Finding a proper default value is hard. The reason I set it 0 is that it seems to be working better for streaming, according to the ticket description. 
I guess, we can set it the same as the previous value (30000) and adjust it on-demand.

Reply:
> Finding a proper default value is hard. 

💯  couldn't agree more

> The reason I set it 0 is that it seems to be working better for streaming

Sure. I'm not fully arguing against a different value. Just trying to reason about the change in behavior. It could even be that we want the streaming timeout with some higher value, but not zero, which might defeat the purpose of the original ticket introducing TCP User Timeout (CASSANDRA-14358).
For the record, [here](https://github.com/torvalds/linux/blob/9ff9b0d392ea08090cd1780fb196f36dbb586529/net/ipv4/tcp_timer.c#L363-L369) is the area of code that made me think it just disables the mechanism altogether. I only point it out to say that I think it's a little different than using some other "system default" as some of the higher level docs say. I'm wondering if the "system default" is to not have this mechanism enabled.

Reply:
It may be interesting to have that discussion on the ticket to get other people attention. Jon might have an opinion as he is the one that opened the ticket.

Reply:
Will take it back to the JIRA ticket and comment there.